homeassistant/components/esphome/manager.py

```diff
@@ -352,6 +389,8 @@ async def _on_connnect(self) -> None:
         cli = self.cli
         stored_device_name = entry.data.get(CONF_DEVICE_NAME)
         unique_id_is_mac_address = unique_id and ":" in unique_id
+        if entry.options.get(CONF_SUBSCRIBE_LOGS):
+            cli.subscribe_logs(self._async_on_log, LogLevel.LOG_LEVEL_VERY_VERBOSE)
```

Comment:
We could optimize this a bit to reduce the log level to whatever _LOGGER level is, and than subscribe to logging changed events and update the device so we reduce the traffic, however this is off by default, and that could come in a followup PR as to not expand the scope here.